src/components/CourseGradeItem.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import { Box, VStack, HStack, Text, Modal, Button, Input } from 'native-base';
+import { useState } from 'react';
+import { TouchableOpacity } from 'react-native';
+import { useDispatch } from 'react-redux';
+import { updateActualGrade } from '../redux/courseSlice';
+import * as SQLite from 'expo-sqlite';
+type CourseGradeItemProps = {
```

Comment:
Move all types to src/types/index.d.ts. Also please notice how I already had a type for Grade.

Reply:
So this part
```
grade: {
    id: number;
    data: {
      name: string;
      actualScore: number | null;
      maxScore: number;
      weight: number;
    };
  };
```
is not necessary